### ISSUE TITLE
Error message displays before form submission

### ISSUE THREAD

**Issue by MaddyDaigle** [GCTC-NTGC/gc-digital-talent#4809]
## Description

The error message currently displays if there are any empty required fields BEFORE the form is submitted.  This causes confusion from a screen reader perspective since the placement of the field is at the top of the page, which the reader is unaware of (and it is constantly changing while the user is working).

## Wanted behaviour
The error message for required fields should only be displayed AFTER the form is submitted.  This allows the screen reader user to see a list of the fields that need to be fixed at the top of the page. 

 After the user attempts to save & the page loads, navigate the user to the top of the "Oops you have some errors" message (focus), so that the error message text is the first thing that is read.

See [WCAG Advisory technique G139](https://www.w3.org/WAI/WCAG21/Techniques/general/G139) for more information on this issue. Notice that, in all examples, the error is only shown after form submission.

## Screenshot
![Capture](https://user-images.githubusercontent.com/1794660/202570344-c880e941-614a-459b-b63e-c138259f719f.PNG)

## Acceptance Criteria

- [x] Only show the big alert at the top of the page after the user hits the submit button. NOT when they're on the page making changes
- [x] Keep in-line errors contained to ONLY that specific field when the user made an error and moves away and not everything on the page at once if the user has errors before hitting the submit button.
- [x] Verify everything is marked all at once AFTER the user clicked submit and has many errors.

**Comment by esizer** [comment]
I believe we should we consider keeping the inline errors during typing. [WAI recommends this as an approach when it is available](https://www.w3.org/WAI/tutorials/forms/notifications/#inline:~:text=Instant%20feedback%20during%20typing%20can%20be%20extremely%20helpful.%20For%20example%2C%20checking%20the%20availability%20of%20a%20username%20in%20the%20previous%20example%20required%20the%20user%20to%20resubmit%20the%20form%20%E2%80%93%20possibly%20multiple%20times.). It can be far better for users, especially with longer forms and lot's of validation to have immediate feedback rather than needing to fix all fields and resubmit the form in order to know if the changes fixed the issues or not.

The only reason WCAG does not outline this is that they expect all examples to work without additional scripting unless necessary.

**Comment by MaddyDaigle** [comment]
That's fair @esizer, as long as the error message isn't shown until after the form is submitted; and the "oops, you have some errors" should be read out. :)

**Comment by Jerryescandon** [comment]
This one is very similar to #4831 But I believe Eric and Maddy already hit the two main things.

- [ ] Only show the big alert at the top of the page after the user hits the submit button. NOT when they're on the page making changes
- [ ] Keep in-line errors contained to ONLY that specific field when the user made an error and moves away and not everything on the page at once if the user has errors before hitting the submit button.
- [ ] Verify everything is marked all at once AFTER the user clicked submit and has many errors.
- [ ] Check this interaction during the adding of skills, cause we expect that to be a funky spot.

**Comment by patcon** [comment]
eric: add alert role back to input error if doing inline. :bulb:
